### PR TITLE
Mongo db integration and testing is done

### PR DESCRIPTION
**Pull request by sgavin0813**
I have migrated to MongoDb and unit tested my code.
It is working Okay.
Changes were made to Core.py classes Wiki and Page class

### CODE REVIEW

File: tests/singh-gavin-unittests/__pycache__/TestPageClass.cpython-311.pyc
Comment (biplovale) [inline tests/singh-gavin-unittests/__pycache__/TestPageClass.cpython-311.pyc:1]: This looks like pycache files. Should be removed.

File: tests/singh-gavin-unittests/__pycache__/TestWikiClass.cpython-311.pyc
Comment (biplovale) [inline tests/singh-gavin-unittests/__pycache__/TestWikiClass.cpython-311.pyc:1]: This looks like pycache files. Should be removed.

File: wiki/__pycache__/DataAccessObject.cpython-311.pyc
Comment (biplovale) [inline wiki/__pycache__/DataAccessObject.cpython-311.pyc:1]: This looks like pycache files. Should be removed.

File: wiki/__pycache__/__init__.cpython-311.pyc
Comment (biplovale) [inline wiki/__pycache__/__init__.cpython-311.pyc:1]: This looks like pycache files. Should be removed.

File: wiki/__pycache__/core.cpython-311.pyc
Comment (biplovale) [inline wiki/__pycache__/core.cpython-311.pyc:1]: This looks like pycache files. Should be removed.

File: wiki/core.py [hunk 1]
```diff
@@ -115,30 +113,29 @@ def process_markdown(self):
...
     def split_raw(self):
         """
             Split text into raw meta and content.
         """
         self.meta_raw, self.markdown = self.pre.split('\n\n', 1)
 
-    def process_meta(self):
-        """
-            Get metadata.
+    def split_raw(self):
```
Comment (biplovale) [inline wiki/core.py:124]: Why is there two split_raw()? Do we now need the old one?

File: wiki/core.py [hunk 2]
```diff
@@ -115,30 +113,29 @@ def process_markdown(self):
...
+        split_result = self.pre.split('\n\n', 1)
+        if len(split_result) == 2:
+            self.meta_raw, self.markdown = split_result
+        else:
+            self.meta_raw = split_result[0]
+            self.markdown = ""  # or some default value
```
Comment (biplovale) [inline wiki/core.py:131]: Let's keep the preexiting comments for process_meta()

File: wiki/core.py [hunk 3]
```diff
@@ -337,41 +328,52 @@ def index_by(self, key):
...
+            pagetags = doc.get("meta", {}).get("tags", "").split(',')
             for tag in pagetags:
                 tag = tag.strip()
-                if tag == '':
-                    continue
-                elif tags.get(tag):
-                    tags[tag].append(page)
-                else:
-                    tags[tag] = [page]
+                if tag:
```
Comment (biplovale) [inline wiki/core.py:435]: Can you explain why is this if condition needed?

File: wiki/web/__pycache__/__init__.cpython-311.pyc
Comment (biplovale) [inline wiki/web/__pycache__/__init__.cpython-311.pyc:1]: Cache files must be removed.

File: wiki/web/__pycache__/forms.cpython-311.pyc
Comment (biplovale) [inline wiki/web/__pycache__/forms.cpython-311.pyc:1]: Cache files must be removed.

File: wiki/web/__pycache__/routes.cpython-311.pyc
Comment (biplovale) [inline wiki/web/__pycache__/routes.cpython-311.pyc:1]: Cache files must be removed.

File: wiki/web/__pycache__/user.cpython-311.pyc
Comment (biplovale) [inline wiki/web/__pycache__/user.cpython-311.pyc:1]: Cache files must be removed.

File: config.py [hunk 4]
```diff
@@ -4,7 +4,9 @@
 TITLE='Riki' 
 HISTORY_SHOW_MAX=30
 PIC_BASE = '/static/content/'
-CONTENT_DIR = '/Users/smcho/Dropbox/NKU/Course/CSC440/project/Riki/content'
-USER_DIR = '/Users/smcho/Dropbox/NKU/Course/CSC440/project/Riki/user'
+CONTENT_DIR = '/Users/gavinsingh/Documents/Riki/content'
+USER_DIR = '/Users/gavinsingh/Documents/Riki/user'
```
Comment (biplovale) [inline config.py]: CONTENT_DIR and USER_DIR can be completely removed now.

File: config.py [hunk 5]
```diff
@@ -4,7 +4,9 @@
...
 NUMBER_OF_HISTORY = 5
 PRIVATE = True
+# connection string to connect to mongoDB
+connection_string = "mongodb+srv://sgavin0813:password_1234@atlascluster.qdshz9s.mongodb.net/"
```
Comment (biplovale) [inline config.py]: Can you make this all Caps

File: wiki/core.py [hunk 6]
```diff
@@ -163,37 +160,54 @@ def process(self):
...
-        self.path = path
+
+    def __init__(self, db, url, new=False):
+        # MongoDB setup
+        # instead of path(where page is stored we have database)
+        self.db = DataAccessObject.db  # same as one in DataAccess class
         self.url = url
+        self.collection = self.db.pages  # 'pages' is the MongoDB collection name
         self._meta = OrderedDict()
+        self.new = new
```
Comment (biplovale) [inline wiki/core.py:192]: I dont think this is necessary. Why do you have this here?

**Comment by biplovale** [comment]
> I have migrated to MongoDb and unit tested my code. It is working Okay. Changes were made to Core.py classes Wiki and Page class

**Review by biplovale — CHANGES_REQUESTED** [review]
Remove content folder
Remove pycache folder and include it in .gitignore

**Review by sgavin0813 — COMMENTED** [review]
Updated the .gitignore file to exclude cache files from the repository.
Removed previously cached files that were present in the repo to maintain cleanliness and manageability.

**Review by biplovale — APPROVED** [review]

**Pull request merged**